.github/workflows/ci.yml

```diff
@@ -0,0 +1,84 @@
+name: CI
+
+on: pull_request
+
+jobs:
+  setup_matrix:
+    name: 'Setup Test Matrix'
+    runs-on: ubuntu-latest
+    outputs:
+      beaker_setfiles: ${{ steps.get-outputs.outputs.beaker_setfiles }}
+      puppet_major_versions: ${{ steps.get-outputs.outputs.puppet_major_versions }}
+      puppet_unit_test_matrix: ${{ steps.get-outputs.outputs.puppet_unit_test_matrix }}
+    env:
+      BUNDLE_WITHOUT: development:test:release
+    steps:
+      - uses: actions/checkout@v2
+      - name: Setup ruby
+        uses: ruby/setup-ruby@v1
+        with:
+          ruby-version: '2.7'
+          bundler-cache: true
+      - name: Run rake validate
+        run: bundle exec rake validate
+      - name: Setup Test Matrix
+        id: get-outputs
+        run: bundle exec metadata2gha --use-fqdn --pidfile-workaround false
+
+  unit:
+    needs: setup_matrix
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        include: ${{fromJson(needs.setup_matrix.outputs.puppet_unit_test_matrix)}}
+    env:
+      BUNDLE_WITHOUT: development:system_tests:release
+      PUPPET_VERSION: "~> ${{ matrix.puppet }}.0"
+    name: Puppet ${{ matrix.puppet }} (Ruby ${{ matrix.ruby }})
+    steps:
+      - uses: actions/checkout@v2
+      - name: Setup ruby
+        uses: ruby/setup-ruby@v1
+        with:
+          ruby-version: ${{ matrix.ruby }}
+          bundler-cache: true
+      - name: Create task helper symlink
+        run: ln -s puppet-gitlab_ci_runner/spec/fixtures/ruby_task_helper ../
+      - name: Run tests
+        run: bundle exec rake
+
+  acceptance:
+    needs: setup_matrix
+    runs-on: ubuntu-latest
+    env:
+      BUNDLE_WITHOUT: development:test:release
+    strategy:
+      fail-fast: false
+      matrix:
+        setfile: ${{fromJson(needs.setup_matrix.outputs.beaker_setfiles)}}
+        puppet: ${{fromJson(needs.setup_matrix.outputs.puppet_major_versions)}}
+    name: ${{ matrix.puppet.name }} - ${{ matrix.setfile.name }}
+    services:
+      gitlab:
+        image: gitlab/gitlab-ce
+        ports:
+          - 80:80
```

Comment:
The Docker health check only waits for the container, it does not check for the API responding on http/80.

https://github.com/voxpupuli/puppet-gitlab_ci_runner/pull/93/checks?check_run_id=1456486776 shows this race condition. 

The Travis configuration called the [setup-gitlab.sh](https://github.com/voxpupuli/puppet-gitlab_ci_runner/blob/master/scripts/start-gitlab.sh#L17) script which invokes an endless waiting loop. 

Probably this can be fixed by adding a new step which calls this script in the steps below, after checking the network settings. cc @bastelfreak

Reply:
@dnsmichi do you have permissions to push to this PR? If so, it's fine to add commits to this PR. We can always clean them up before we merge.

Reply:
@ekohl I'm testing my assumption in a fork currently. Thanks for the poke, I should actually have permissions :)

Reply:
Seems I was wrong, it is just the network and the DNS record just not exposed by the `services` setting in GitHub actions. 

According to the docs it _should_ work though: https://docs.github.com/en/free-pro-team@latest/actions/guides/about-service-containers#running-jobs-in-a-container

Reply:
That was also my impression. The only thing I can think of is that the services and what beaker-docker launches uses different networks. Note that the DNS could not be resolved (I tried without the IP in hosts hack) which may also indicate that they're isolated.

Reply:
I did not know that the acceptance tests are run in Docker by beaker, but I just remembered a similar chat with @davids about Gitpod, Docker and acceptance tests. 

https://github.com/voxpupuli/voxpupuli-acceptance

I'd guess you need to manually start the Docker container for gitlab/gitlab-ce then, and build your own isolated network. Or there is a way to "pre start" the beaker container with GitHub actions. 

The docs and community threads advise against docker-compose and own setups though, maybe this scenario won't work with actions and the services setting.

Reply:
One solution might be to launch a gitlab-ce container via beaker-docker so they end up in the same network, like https://github.com/voxpupuli/puppet-vault_lookup/blob/master/spec/acceptance/nodesets/docker/docker.yml does.